packages/jellyfish-transaction-builder/src/txn/txn_builder.ts

```diff
@@ -98,15 +100,21 @@ export abstract class P2WPKHTxnBuilder {
    * @param {OP_DEFI_TX} opDeFiTx to create
    * @param {Script} changeScript to send unspent to after deducting the fees
    * @param {BigNumber} [outValue=0] for the opDeFiTx, usually always be 0.
+   * @param {Array<{ vin: Vin, vout: Vout, prevout: Prevout }>} [customVinVout = []] for custom vin and vout
    */
   async createDeFiTx (
     opDeFiTx: OP_DEFI_TX,
     changeScript: Script,
-    outValue: BigNumber = new BigNumber('0')
+    outValue: BigNumber = new BigNumber('0'),
+    customVinVout: Array<{ vin: Vin, vout: Vout, prevout: Prevout }> = []
```

Comment:
Can we move this feature out to another PR? Since it can be separated from `updateMasternode` to reduce the impact and properly contextualize the changes.